x-pack/legacy/plugins/siem/public/components/link_to/redirect_to_detection_engine.tsx

```diff
@@ -63,9 +63,10 @@ export const RedirectToEditRulePage = ({
 
 const baseDetectionEngineUrl = `#/link-to/${DETECTION_ENGINE_PAGE_NAME}`;
 
-export const getDetectionEngineUrl = () => `${baseDetectionEngineUrl}`;
-export const getDetectionEngineAlertUrl = () =>
-  `${baseDetectionEngineUrl}/${DetectionEngineTab.alerts}`;
+export const getDetectionEngineUrl = (search?: string) =>
+  `${baseDetectionEngineUrl}${search != null ? `?${search}` : ''}`;
```

Comment:
Consider creating a utility function for `${search != null ? `?${search}` : ''}`, because there are six instances of it in this PR, and likely additional instances in the future

Reply:
I'm not sure if it's possible / valid to change the type signatures of the functions that take `(search?: string)` to something like `(search: string | null)`, but an advantage of such a change is the type checker would force the caller to always handle this case by passing `search` or an explicit `null` when `search` should not be provided. 

For example, making this parameter non-optional makes the type checker flag the following code in `x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/index.tsx`:

```
        <DetectionEngineHeaderPage
          backOptions={{
            href: getDetectionEngineUrl(),
            text: i18n.BACK_TO_DETECTION_ENGINE,
          }}
          title={i18n.PAGE_TITLE}
        >
```

* If the above code really should be passing `search`, then the type checker just found a bug
* If the above code is a valid case to _not_ pass `search`, we would be forced to explicitly pass `null`, which feels OK to me
* Making `search` non-optional also causes the type checker to flag `getBreadcrumbs` in `x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/utils.ts` because it uses a _slight_ variation of the "search checking pattern", where in this case it's using `search[0]` instead:
```
${!isEmpty(search[0]) ? search[0] : ''}
```

Reply:
That's why I did it like that because I just wanted to focus this PR on the overview page. Also, I remember that we made a decision that we did not want the breadcrumb to follow the url state for some reason. (maybe we can discuss it in one of our meetings)

Reply:
but I mean you are correct, we should have a follow-up PRs to get all link through the app corrected and discussed breadcrumb.

Reply:
It looks like all breadcrumbs have the URL state, so that resolved the question above. I will go over in another PR to make everything following the same pattern. so the breadcrumb's URL can be also simplified from this PR.